cpp/src/Ice/SSL/SSLUtil.h

```diff
@@ -44,7 +45,7 @@ namespace Ice::SSL
     void readFile(const std::string&, std::vector<char>&);
 
     // Determine if a file or directory exists, with an optional default directory.
-    bool checkPath(const std::string&, const std::string&, bool, std::string&);
+    std::optional<std::string> checkPath(const std::string&, const std::string&, bool);
```

Comment:
Maybe this function should be named `resolvePath` and the second parameter should be an `optional<string> parentDir`.

Reply:
Split the function in two `resolveFilePath` and `resolveDirPath`, I keep an `std::string` for parent directory parameter because we always pass a value, and passing an optional would mean to move this empty check to all the callers.